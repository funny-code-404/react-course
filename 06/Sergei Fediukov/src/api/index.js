export const baseUrl = 'https://private-anon-8de7e119e1-carsapi1.apiary-mock.com/cars';

export const defaultConfig = {
    headers: {
        Accept: 'application/json',
    }
}

export const API = {
    request(url, config = {}) {
        return fetch(url, {
            ...defaultConfig,
            ...config,
        });
    },
    get(config = {}) {
        return this.request(`${baseUrl}`, config);
    },
    // post(path, config = {}) { },
    // put(path, config = {}) { },
    // patch(path, config = {}) { },
    // delete(path, config = {}) { },
};
