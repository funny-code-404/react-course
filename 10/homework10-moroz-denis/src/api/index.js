export const baseUrl = 'https://age-of-empires-2-api.herokuapp.com/api/v1';

export const defaultConfig = {
    headers: {
        Accept: 'application/json',
    }
}

export const API = {
    request(url, config = {}) {
        return fetch(url, {
            ...defaultConfig,
            ...config,
        });
    },
    get(path, config = {}) {
        return this.request(`${baseUrl}${path}`, config);
    },
    post(path, config = {}) {},
    put(path, config = {}) {},
    patch(path, config = {}) {},
    delete(path, config = {}) {},
};