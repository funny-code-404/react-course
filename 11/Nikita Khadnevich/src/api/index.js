export const baseUrl = 'https://jsonplaceholder.typicode.com';

export const defaultConfig = {
  headers: {
    Accept: 'application/json',
  },
};

export const API = {
  request(url, config = {}) {
    return fetch(url, {
      ...defaultConfig,
      config,
    });
  },
  get(path) {
    return this.request(`${baseUrl}${path}`);
  },
};
