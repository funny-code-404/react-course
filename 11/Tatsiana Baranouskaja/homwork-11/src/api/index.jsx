const api = {
    request: async (url, params = {}) => {
        const res = await fetch(url, params);
        const data = await res.json();


        return data;
    },
    get: async (url) => {
        const data = await api.request(url);

        return data;
    },
    post: async (url, params) => {
        const data = await api.request(url, {
            metod: 'POST',
            ...params,
        });

        return data;
    },
    put: () => {},
    delete: () => {},
};

export const userApi = {
    getUsers: async (url) => {
        return api.get(url);
    }
};

export const API = {
    userApi,
};