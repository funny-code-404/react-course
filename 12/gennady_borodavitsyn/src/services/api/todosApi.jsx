import createInstance from '../instance';
import { getTodoListFromServer } from '../utils/mapping/dataFromServer';

const API = createInstance();

export const getTodoList = async (config = {}) => {
  const res = await API.get('/items.json', config);

  return getTodoListFromServer(res);
};

export const addTodo = async (body) => {
  await API.post('/items', body);
  const res = await API.get('/items.json');

  return getTodoListFromServer(res);
};

export const deleteTodo = async (id) => {
  await API.delete(`/items/${id}.json`);
  const res = await API.get('/items.json');

  return getTodoListFromServer(res);
};

export const updateTodo = async ({ id, title }) => {
  await API.patch(`/items/${id}`, { title: title });
  const res = await API.get('/items.json');

  return getTodoListFromServer(res);
};
