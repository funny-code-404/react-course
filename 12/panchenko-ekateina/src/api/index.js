export const baseUrl = "https://age-of-empires-2-api.herokuapp.com/api/v1";

export const getFormatedPath = (path) =>
  path.replace(baseUrl, "").toLowerCase();

export const getFormatedName = (path) => {
  const splitedPath = path.split("/");

  return splitedPath[splitedPath.length - 1].replace("_", " ");
};

export const defaultConfig = {
  headers: {
    Accept: "application/json",
  },
};

export const API = {
  request(url, config = {}) {
    return fetch(url, {
      ...defaultConfig,
      config,
    });
  },
  get(path) {
    return this.request(path);
  },
};
