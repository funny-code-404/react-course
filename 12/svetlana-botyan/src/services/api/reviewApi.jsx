import createInstance from '../instanse';
import { getReviewrsListFromServer } from '../utils/mapping/dataFromServer';

const API = createInstance();

export const getAllReviews = async (config = {}) => {
  const response = await API.get('/reviews', config);

  return getReviewrsListFromServer(response);
};

export const addReview = async (obj) => {
  await API.post('/reviews', obj);

  const response = await API.get('/reviews');
  return getReviewrsListFromServer(response);
};

export const deleteReview = async (obj) => {
  await API.delete('/reviews', { data: obj });

  const response = await API.get('/reviews');
  return getReviewrsListFromServer(response);
};

export const updateReview = async (obj) => {
  await API.patch('/reviews', obj);

  const response = await API.get('/reviews');
  return getReviewrsListFromServer(response);
};
