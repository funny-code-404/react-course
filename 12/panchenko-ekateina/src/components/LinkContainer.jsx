import styled from "styled-components";
import { Link } from "react-router-dom";
import { getFormatedName, getFormatedPath, baseUrl } from "../api";
import { memo } from "react";

const StyledLink = styled(Link)`
  text-transform: capitalize;
  color: #fff;
  cursor: pointer;
`;

export const LinkContainer = memo(({ title, link, type }) => (
  <p>
    {title}:{" "}
    {typeof link === "string" ? (
      <>
        {link.includes(baseUrl) ? (
          <StyledLink
            to={{
              pathname: getFormatedPath(link),
              state: {
                link,
              },
            }}
          >
            {getFormatedName(link, type)}
          </StyledLink>
        ) : (
          <span>{link}</span>
        )}
      </>
    ) : (
      <>
        {link.map((item, index) =>
          item.includes(baseUrl) ? (
            <StyledLink
              key={`${item}_${index}`}
              to={{
                pathname: getFormatedPath(item),
                state: {
                  link: item,
                },
              }}
            >
              {getFormatedName(item, type)}
              {index !== link.length - 1 ? ", " : ""}
            </StyledLink>
          ) : (
            <span key={`${item}_${index}`}>
              {item}
              {index !== link.length - 1 ? ", " : ""}
            </span>
          )
        )}
      </>
    )}
  </p>
));
