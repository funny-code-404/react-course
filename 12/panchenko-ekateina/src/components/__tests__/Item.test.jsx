import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { Item } from "..";
import { LinkContainer } from "../LinkContainer";
import { baseUrl } from "../../api";
import "jest-styled-components";

configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  NavLink: "a",
  Link: "a",
  Route: ({ children, path }) => children({ match: path === "/" }),
}));
jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
}));

describe("Item mounting behavior", () => {
  const props = {
    location: {
      state: {
        data: {
          name: "Test",
          expansion: "Test expansion",
          civilization_bonus: "Test civilization bonus",
          attack_bonus: ["Attack bonus 1", "Attack bonus 2"],
          develops_in: [`${baseUrl}/one`, `${baseUrl}/two`],
        },
      },
    },
  };
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Item {...props} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 1 component snapshot", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("should render 'Item' component with actual prop", () => {
    expect(wrapper.find(LinkContainer)).toHaveLength(1);
    expect(wrapper.find(LinkContainer).props().title).toEqual("Develops in");
  });
});
