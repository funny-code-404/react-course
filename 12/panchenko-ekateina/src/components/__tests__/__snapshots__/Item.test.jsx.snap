// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item mounting behavior should match snapshot 1`] = `
.c0 {
  background-image: url(ageOfEmpirescivilization.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  min-height: 100vh;
}

.c1 {
  background-color: rgba(0,0,0,0.6);
  min-height: 100vh;
  padding-top: 150px;
}

.c4 {
  text-transform: capitalize;
  color: #fff;
  cursor: pointer;
}

.c2 {
  background: #000;
  color: #e5b378;
  font-size: 25px;
  margin: auto;
  padding: 50px;
  width: 35%;
}

.c3 {
  list-style: none;
  padding: 30px;
}

@media (max-width:780px) {
  .c1 {
    padding-top: 240px;
  }
}

@media (max-width:780px) {
  .c2 {
    width: 90%;
    font-size: 20px;
  }
}

<Item
  location={
    Object {
      "state": Object {
        "data": Object {
          "attack_bonus": Array [
            "Attack bonus 1",
            "Attack bonus 2",
          ],
          "civilization_bonus": "Test civilization bonus",
          "develops_in": Array [
            "https://age-of-empires-2-api.herokuapp.com/api/v1/one",
            "https://age-of-empires-2-api.herokuapp.com/api/v1/two",
          ],
          "expansion": "Test expansion",
          "name": "Test",
        },
      },
    }
  }
>
  <div>
    <styled.div
      background="ageOfEmpirescivilization.jpg"
    >
      <div
        className="c0"
      >
        <styled.div>
          <div
            className="c1"
          >
            <styled.div>
              <div
                className="c2"
              >
                <p>
                  Name: 
                  Test
                </p>
                <p>
                  Expansion: 
                  Test expansion
                </p>
                <div>
                  Attack bonus
                  :
                   
                  <styled.ul>
                    <ul
                      className="c3"
                    >
                      <li
                        key="Attack bonus 1_0"
                      >
                        Attack bonus 1
                      </li>
                      <li
                        key="Attack bonus 2_1"
                      >
                        Attack bonus 2
                      </li>
                    </ul>
                  </styled.ul>
                </div>
                <div>
                  Civilization bonus
                  :
                   
                  <p>
                    Test civilization bonus
                  </p>
                </div>
                <Memo()
                  link={
                    Array [
                      "https://age-of-empires-2-api.herokuapp.com/api/v1/one",
                      "https://age-of-empires-2-api.herokuapp.com/api/v1/two",
                    ]
                  }
                  title="Develops in"
                >
                  <p>
                    Develops in
                    :
                     
                    <styled.a
                      key="https://age-of-empires-2-api.herokuapp.com/api/v1/one_0"
                      to={
                        Object {
                          "pathname": "/one",
                          "state": Object {
                            "link": "https://age-of-empires-2-api.herokuapp.com/api/v1/one",
                          },
                        }
                      }
                    >
                      <a
                        className="c4"
                        to={
                          Object {
                            "pathname": "/one",
                            "state": Object {
                              "link": "https://age-of-empires-2-api.herokuapp.com/api/v1/one",
                            },
                          }
                        }
                      >
                        one
                        , 
                      </a>
                    </styled.a>
                    <styled.a
                      key="https://age-of-empires-2-api.herokuapp.com/api/v1/two_1"
                      to={
                        Object {
                          "pathname": "/two",
                          "state": Object {
                            "link": "https://age-of-empires-2-api.herokuapp.com/api/v1/two",
                          },
                        }
                      }
                    >
                      <a
                        className="c4"
                        to={
                          Object {
                            "pathname": "/two",
                            "state": Object {
                              "link": "https://age-of-empires-2-api.herokuapp.com/api/v1/two",
                            },
                          }
                        }
                      >
                        two
                      </a>
                    </styled.a>
                  </p>
                </Memo()>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </div>
</Item>
`;
